### DISCUSSION TITLE
benchmark against cacheguard

### DISCUSSION THREAD

**Discussion by osevan** [dragonflydb/dragonfly#299] [Ideas]
https://github.com/danielealbano/cachegrand

Thanks and 

Best regards

**Comment by romange** [comment]
Cache Grand does not position himself as Redis replacement. It's quite possible to introduce fast data structures that will support single key operations. The question is how you solve multi key commands, lua scripts, transactions and snapshotting. 

Dragonfly provides a holistic solution for all redis operations and as such is not comparable directly with cachegrind.

**Reply by osevan** [reply]
Ok.

**Reply by danielealbano** [reply]
cachegrand position itself as Redis-compatible, with the goal (looking at the features) to provide Redis enterprise features (and more) and being open source under a very open license. 

The two platforms are comparable when running under the same conditions, of course comparing the on-disk backend of cachegrand with the in-memory db of dragonflydb wouldn't make sense but cachegrand can run in memory as well. 
cachegrand is also currently missing a lot of commands, we are rewriting them from scratch, what can be certainly compared are the classinc GET, SET, etc without the cache mode on.

Functionalities like multiple keys and transactions are already in place (the command MULTI is per spec not a "classic" transaction but a sequence of commands executed in sequence), snapshotting and Lua support will be implemented when we will get there :)

In general however benchmarks should always be interpreted for what they are: the maximum that can be achieved under very simple circumstances.
